src/Playwright/Transport/Channels/BrowserChannel.cs

```diff
@@ -103,7 +103,7 @@ internal override void OnMessage(string method, JsonElement? serverParams)
 
     internal Task<CDPChannel> NewBrowserCDPSessionAsync()
     => Connection.SendMessageToServerAsync<CDPChannel>(
```

Comment:
Would it make sense to add Channel.SendMessageToServerAsync and call it in all these places? It then can check if the channel has been disposed already.